### ISSUE TITLE
Get last emitted value after subscription

### ISSUE THREAD

**Issue by unlight** [a-jie/RxEmitter#1]
```
const RxEmitter = require('rxemitter').RxEmitter;
RxEmitter.emit("ADD_AN_NUMBER", 1);
RxEmitter.emit("ADD_AN_NUMBER", 2);
RxEmitter.on('ADD_AN_NUMBER').subscribe(x=> console.log(x));
```
Prints nothing.
Is possible to get last value after subscription?
So expected behavior is printing 2.

**Comment by unlight** [comment]
Forked [unlight/RxEmitter](https://github.com/unlight/RxEmitter)

**Comment by a-jie** [comment]
The order of registration.
Your advice is good, I have to think about how to achieve. thank you very much

**Comment by a-jie** [comment]
The order of registration.
Your advice is good, I have to think about how to achieve. thank you very much

**Comment by unlight** [comment]
What is the status/resolution?

**Comment by unlight** [comment]
What is the status/resolution?

**Comment by a-jie** [comment]
Sorry, I haven’t come and implemented the logic here. I'm really busy lately, sorry

**Comment by Hezard** [comment]
To manage this behavior you should replace Subject with BehaviorSubject:

`
        if (!this.cache[eventName].subject) {
            this.cache[eventName].subject = new BehaviorSubject(null);
        }
`

**Comment by unlight** [comment]
@Hezard 
Yes, and additionally we need add `.skip(1)` in `on()` method if subject already exists.
See my forked branch https://github.com/unlight/RxEmitter
